provider/infoblox/infoblox.go

```diff
@@ -228,8 +242,44 @@ func (p *InfobloxProvider) Records(ctx context.Context) (endpoints []*endpoint.E
 			if _, err := strconv.Unquote(res.Text); err != nil {
 				res.Text = strconv.Quote(res.Text)
 			}
+			logrus.Debugf("Record='%s' TXT:'%s'", res.Name, res.Text)
 			endpoints = append(endpoints, endpoint.NewEndpoint(res.Name, endpoint.RecordTypeTXT, res.Text))
 		}
+
+		// Concatenate A-records with same dnsname
+		sort.Sort(byDNSName(endpointsTypeA))
+		idc := int(-1)
+		for id1, a := range endpointsTypeA {
+			if id1 > idc {
+				var targets endpoint.Targets
+				for _, target := range a.Targets {
+					logrus.Debugf("DEBUG New Record:'%d':'%s':'%d'-'%s':'%s'", id1, "-", idc, a.DNSName, target)
+					if len(targets) == 0 {
+						targets = endpoint.NewTargets(target)
+					} else {
+						targets = append(targets, target)
+					}
+				}
+				for id2, b := range endpointsTypeA {
+					if id2 > id1 && a.DNSName == b.DNSName {
+						for _, target := range b.Targets {
+							logrus.Debugf("DEBUG Add Record:'%d':'%d':'%d'-'%s':'%s'", id1, id2, idc, a.DNSName, target)
+							targets = append(targets, target)
+						}
+						idc = id2
+					}
+				}
+				if idc < id1 {
+					idc = id1
+				}
```

Comment:
Is there any way we can rewrite this to be more readable? It's a lot of work with indexes of arrays which is hard to read and possibly error prone to maintain.

Reply:
Like:
```golang
// Concatenate A-records with same dnsname
sort.Sort(byDNSName(endpointsTypeA))
var runner *endpoint.Endpoint
for _, endpoint := range endpointsTypeA {
  if runner == nil || endpoint.DNSName != runner.DNSName {
    // add unique to collection
    endpoints = append(endpoints, endpoint)
    runner = endpoint
  } else {
    // add targets of double entries and sort it
    runner.Targets = append(runner.Targets, endpoint.Targets...)
    sort.Sort(runner.Targets) // Dont know if this is needed?
  }
}
```
And then also remove:
```golang
// logrus.Debugf("List Endpoints, after concatenate and sort of A-records")
for _, ep := range endpoints {
    sort.Sort(ep.Targets)
    // logrus.Debugf("Record='%s' '%s':'%s'", ep.DNSName, ep.RecordType, ep.Targets)
}
```

Reply:
Tnxs Bob for helping out. Looks very cool. Already committed your suggestions. Hope this PR will finally get merged.

Reply:
@Raffo Bob helped rewriting this part.  I think its a very cool piece of code now, how do you think?